### ISSUE TITLE
Dependency check causes pytest failures when running individual tests.

### ISSUE THREAD

**Issue by donaldcampbelljr** [pepkit/pipestat#181]
Using a complex output schema:

```
title: An example Pipestat output schema
description: A pipeline that uses pipestat to report sample and project level results.
type: object
properties:
  pipeline_name: "default_pipeline_name"
  samples:
    type: object
    properties:
      collection_of_images:
        description: "This store collection of values or objects"
        type: array
        items:
          properties:
              prop1:
                description: "This is an example file"
                $ref: "#/$defs/file"
$defs:
  file:
    type: object
    object_type: file
    properties:
      path:
        type: string
      title:
        type: string
    required:
      - path
      - title

```

and attempting to report a value:
```

                "collection_of_images": [
                    {
                        "items": {
                            "properties": {
                                "prop1": {
                                    "properties": {
                                        "path": "pathstring",
                                        "title": "titlestring",
                                    }
                                }
                            }
                        }
                    }
                ]
            },
```

I see an error during testing:
```
        except e._NO_TRACEBACK as ex:
>           raise ex.with_traceback(None)
E           psycopg.errors.UndefinedColumn: column default_pipeline_name__sample.collection_of_images does not exist
E           LINE 1: ...itch_value, default_pipeline_name__sample.md5sum, default_pi...
E                                                                        ^

venv/lib/python3.10/site-packages/psycopg/cursor.py:732: UndefinedColumn
```

Double checking the db, it appears that column is indeed not being created during table creation.

**Comment by donaldcampbelljr** [comment]
We currently have the field definition as a `list[dict]`:
![image](https://github.com/pepkit/pipestat/assets/125581724/03008d88-b63f-447b-8771-bb20d9b293d9)

**Comment by donaldcampbelljr** [comment]
Ok, I think there might be something else going on here. If I remove the collection of images from my test and the recursive schema, I get the same error but for the next complex object, e.g `output_file`.

**Comment by donaldcampbelljr** [comment]
Oh, its our check to determine if the dependencies are satisfied. It creates an initial table using a _different_ schema than the test:
`pipestat report --c 'tests/data/config.yaml' -i 'name_of_something' -v 'test_value' -r 'dependency_value'"`

So, this causes issues with the very first test IF you are running them individually. I actually ran into this issue last week and did not realize/remember I had reverted the checks back to their original state.
